src/pages/index.js

```diff
@@ -61,66 +76,124 @@ const addPlacePopup = new PopupWithForm(
   placeModalSelector,
   handleAddPlaceFormSubmit
 );
-
 addPlacePopup.setEventListeners();
 
-function addPlace() {
-  formValidators["place-form"].resetValidation();
-  addPlacePopup.open();
-}
+const editAvatarPopup = new PopupWithForm(
+  editAvatarModalSelector,
+  handleEditAvatarSubmit
+);
+editAvatarPopup.setEventListeners();
+const editAvatarForm = editAvatarPopup.getForm();
+const avatarUrlInput = editAvatarForm.querySelector("[name='url']");
 
-const userInfo = new UserInfo({
-  nameSelector: profileNameSelector,
-  jobSelector: profileJobSelector,
-});
+const imagePopup = new PopupWithImage(imageModalSelector);
+imagePopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithConfirmation(
+  deleteCardModalSelector,
+  deleteCard
+);
+deleteCardPopup.setEventListeners();
 
 function editProfile() {
-  const profileInfo = userInfo.getUserInfo();
-  const { name, job } = profileInfo;
+  const { name, job } = userInfo.getUserInfo();
   nameInput.value = name;
   jobInput.value = job;
   profilePopup.open();
 }
 
 function handleProfileFormSubmit({ name, job }) {
-  userInfo.setUserInfo({ name, job });
-  profilePopup.close();
-  formValidators["profile-form"].resetValidation();
+  profilePopup.setButtonText("Saving...");
+  api
+    .editUserInfo({ name, job })
+    .then(() => {
+      userInfo.setUserInfo({ name, job });
+      profilePopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profilePopup.setButtonText("Save");
+      formValidators["profile-form"].resetValidation();
+    });
+}
+
+function addPlace() {
+  formValidators["place-form"].resetValidation();
+  addPlacePopup.open();
 }
 
 function handleAddPlaceFormSubmit({ title, url }) {
+  addPlacePopup.setButtonText("Saving...");
   const newCard = { name: title, link: url };
-  const cardElement = getCardElement(newCard);
-
-  cardsList.addItem(cardElement, { method: "prepend" });
-
-  addPlacePopup.close();
+  api
+    .addCard(newCard)
+    .then((savedCard) => {
+      cardsList.renderItems([savedCard]);
+      addPlacePopup.close();
+    })
+    .catch((err) => {
+      console.error("Error adding place:", err);
+    })
+    .finally(() => {
+      addPlacePopup.setButtonText("Create");
+      formValidators["place-form"].toggleButtonState();
```

Comment:
`      formValidators["place-form"].toggleButtonState();
`

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.